kube/src/client/mod.rs

```diff
@@ -390,7 +432,110 @@ impl From<Config> for reqwest::ClientBuilder {
         builder = builder.default_headers(config.headers);
         builder = builder.danger_accept_invalid_certs(config.accept_invalid_certs);
 
-        builder
+        Ok(builder)
+    }
+}
+
+#[cfg(feature = "native-tls")]
+impl TryFrom<Config> for AsyncTlsConnector {
+    type Error = Error;
+    fn try_from(config: Config) -> Result<Self> {
+        let mut builder = TlsConnector::builder();
+        if let Some((identity, identity_password)) = config.identity.as_ref() {
+            builder.identity(
+                Identity::from_pkcs12(identity, identity_password)
+                    .map_err(|e| Error::SslError(format!("{}", e)))?,
+            );
+        }
+        if let Some(ders) = config.root_cert {
+            for der in ders {
+                builder.add_root_certificate(
+                    Certificate::from_der(&der.0).map_err(|e| Error::SslError(format!("{}", e)))?,
+                );
+            }
+        }
+        if config.accept_invalid_certs {
+            builder.danger_accept_invalid_certs(config.accept_invalid_certs);
+        }
+        let connector = builder.build().map_err(|e| Error::SslError(format!("{}", e)))?;
+        Ok(AsyncTlsConnector::from(connector))
+    }
+}
+
+#[cfg(feature = "rustls-tls")]
+impl TryFrom<Config> for AsyncTlsConnector {
```

Comment:
oh, wow you've done all the tls work as well!

Reply:
I just stole `rustls` pieces from `reqwest` and put it together :see_no_evil: It's basically repeating the work because it's not possible to access the inner data. I wish there was a way to reuse TLS connector on both clients.

Reply:
If we have to do all the TLS stuff ourselves then I'm almost starting to wonder how much we gain from Reqwest over plain Hyper anyway..

Reply:
interesting point. we do have most of their dependencies in Cargo.toml, except the gzip/async-compression stuff AFAIKT. Would be interesting to see how much it introduces to kube to lift hyper directly up as you say. It was already a bit of a dance to get it to handle certs in way reqwests expected.

Reply:
That is very much peripheral to this issue though. We could open one about it to potentially replace #100 ?

Reply:
Sure @clux

Reply:
#362